modules/ws/src/main/java/org/mule/module/ws/consumer/WSConsumer.java

```diff
@@ -107,20 +105,30 @@ public MuleEvent process(MuleEvent event) throws MuleException
     }
 
 
-    private void initializeWsdlRetrievalStrategy()
+    private void initializeWsdlRetrievalStrategy() throws InitialisationException
     {
         HttpRequesterConfig httpRequesterConfig = config.getConnectorConfig();
+        String url = config.getWsdlLocation();
 
-        if (httpRequesterConfig == null || httpRequesterConfig.getProxyConfig() == null)
+        boolean isHttpRequester = url.startsWith(HttpConnector.HTTP);
```

Comment:
This doesn't check the protocol part of the URL: detects http:// but also http.wsdl which is a file.